### PR TITLE
feat: uploadprocessor grb

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs [hunk 1]
```diff
@@ -1,20 +1,167 @@
...
+                {
+                    continue;
+                }
+
+                // If so, update ticket status and job status => preparing
+                await _ticketing.Pending(job.TicketId!.Value, stoppingToken);
+                job.Status = JobStatus.Preparing;
+                await _buildingGrbContext.SaveChangesAsync(stoppingToken);
+
+                var blobObject = await _blobClient.GetBlobAsync(jobName, stoppingToken);
```
Comment (jvandaal) [inline src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs:75]: Should we move this before setting the job/ticket to preparing/pending?

File: src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs [hunk 2]
```diff
@@ -1,20 +1,167 @@
...
+                    continue;
+                }
+
+                try
+                {
+                    // extract, verify, and store data as job records
+                    await using var stream = await blobObject.OpenAsync(stoppingToken);
+                    using var archive = new ZipArchive(stream, ZipArchiveMode.Read, false);
+
+                    var archiveValidator = new ZipArchiveValidator(GrbArchiveEntryStructure);
```
Comment (jvandaal) [inline src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs:88]: Can we inject the ZipArchiveValidator and the ZipArchiveTranslator through an interfaces? Then we can unit test the flows:
* Are all returned job records saved?
* Is the job status set to Error when a problem is returned from the validator?
* Is the ticket set tot Error when a problem is returned from the validator?

File: src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs [hunk 3]
```diff
@@ -1,20 +1,167 @@
...
+                if (blobObject is null)
+                {
+                    _logger.LogError($"No blob found with name: {job.BlobName}");
+                    continue;
+                }
+
+                try
+                {
+                    // extract, verify, and store data as job records
+                    await using var stream = await blobObject.OpenAsync(stoppingToken);
```
Comment (jvandaal) [inline src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs:85]: I'd gather all logic related to 
* checking if the blob is present,
* getting the blobObject
* and opening the blob object
in one place.

File: src/BuildingRegistry.Grb.Processor.Upload/ECSTaskOptions.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿namespace BuildingRegistry.Grb.Processor.Upload
+{
+    public class ECSTaskOptions
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/ECSTaskOptions.cs:3]: Rename to `EcsTaskOptions`

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/Core/FileProblem.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                   && string.Equals(Reason, other.Reason)
+                   && Parameters.SequenceEqual(other.Parameters);
+        }
+
+        public int GetHashCode(FileProblem obj)
+        {
+            return HashCode.Combine(obj.File, obj.Reason, obj.Parameters);
+        }
+
+        public override bool Equals(object obj)
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/Core/FileProblem.cs:49]: Can this method be moved up?
Then all `Equals` methods and `GetHashCode` methods are bunched together

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/ZipArchiveProblems.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        {
+            if (problems == null) throw new ArgumentNullException(nameof(problems));
+
+            return new ZipArchiveProblems(_problems.AddRange(problems));
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is ZipArchiveProblems other && Equals(other);
+        }
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/ZipArchiveProblems.cs:55]: Can you move this to the other `Equals` method?

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderFormatException.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿namespace BuildingRegistry.Grb.Processor.Upload.Zip.Exceptions
+{
+    using System;
+
+    public class DbaseHeaderFormatException : Exception
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderFormatException.cs:5]: Implement `ISerializable` for all exceptions

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/Core/ProblemParameter.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            return other != null
+                   && string.Equals(Name, other.Name)
+                   && string.Equals(Value, other.Value);
+        }
+
+        public int GetHashCode(ProblemParameter obj)
+        {
+            return HashCode.Combine(obj.Name, obj.Value);
+        }
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/Core/ProblemParameter.cs:41]: Move this one down so they are grouped.

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderSchemaMismatchException.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+namespace BuildingRegistry.Grb.Processor.Upload.Zip.Exceptions;
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderSchemaMismatchException.cs:1]: Blocked namespace, do it for the whole project to be sure.

File: src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderSchemaMismatchException.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+using System;
+using System.Runtime.Serialization;
+
+public class DbaseHeaderSchemaMismatchException : Exception
```
Comment (ArneD) [inline src/BuildingRegistry.Grb.Processor.Upload/Zip/Exceptions/DbaseHeaderSchemaMismatchException.cs:6]: See our Domain Exceptions.
Sealed, `[Serializable]` attribute.

**Review by ArneD — APPROVED** [review]

**Pull request merged**